ERCS/erc-7584.md

```diff
@@ -0,0 +1,402 @@
+			   
+   - To convert between *bytes32* and *string* you must use the following methodology:
+```
+   // Converts a bytes32 value to a string
+   
+   // @param {string} _hex A hexadecimal string representation of a bytes32 value
+   // @returns {string} The string representation of the bytes32 value
+   
+     bytes32ToString(_hex) {
+    return this.ethers.utils.toUtf8String(this.ethers.utils.arrayify(_hex).filter(n => n != 0));
+	  }
+```
+```
+  // Converts a string to a bytes32 value
+  
+  // @param {string} _string The string to convert
+  // @returns {string} The bytes32 value
+  
+    stringToBytes32(_string) {
+    let result = this.ethers.utils.hexlify(this.ethers.utils.toUtf8Bytes(_string));
+    while (result.length < 66) {
+      result += '0';
+    }
+    if (result.length !== 66) {
+      throw new Error("invalid web3 implicit bytes32");
+    }
+    return result;
+	}
+  ```
+  - This can also more easily be accomplished using our npm package [@linagee/lnr-ethers](https://www.npmjs.com/package/@linagee/lnr-ethers).
+
+4. **Normalization Standard (ENSIP-15):**
+   - LNR adheres to the [ENSIP-15](https://docs.ens.domains/ens-improvement-proposals/ensip-15-normalization-standard) Normalization Standard, which defines a set of rules for normalizing names. This standard ensures that all registered names are treated consistently, minimizing potential ambiguities and conflicts. 
+   - The ENSIP-15 Normalization Standard encompasses processes such as converting names to lowERCase, removing leading and trailing whitespace, and performing Unicode normalization. This normalization process guarantees that registered names are in their most standardized and compatible form for effective use within the Linagee Name Registrar ecosystem.
+   - Note that *normalization* is enforced by javascript libraries, not by the contract. Linagee encourages normalization by only formally allowing normalized names to be resolved to addresses if using our website for domain name management.
+   - Examples of both **normalized** and **non-normalized** names:
+	   - Normalized✅:
+		   - string: linagee🌎.og
+		   - bytecode: 0x6c696e61676565f09f8c8e000000000000000000000000000000000000000000
+	  - Not normalized❌:
+	       - string: LinagEe🌎.og
+		   - bytecode: 0x4c696e61674565f09f8c8e000000000000000000000000000000000000000000
+	   - Not normalized❌:
+		   - string: linagee🌎.og
+		   - bytecode: 0x6c696e61676565f09f8c8e000000000000000000000000000000000000001a1a
+	   - Not normalized❌:
+		   - string: lina.gee.og
+		   - bytecode: 0x6c696e612e676565000000000000000000000000000000000000000000000000
+
+
+
+## Rationale
+
+The integration of on-chain permanent domain names as both address mappings and CDNs aligns seamlessly with ERC-4804, offering a trustless mapping solution that not only enhances data permanence but also promotes decentralized content delivery, ensuring a robust and censorship-resistant infrastructure for the Ethereum ecosystem.
```

Comment:
This text belongs in the Motivation section. The Motivation section justifies the proposal as a whole, while the Rationale section is supposed to justify technical decisions made within the proposal.

My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We chose to paint the shed red because...